### PR TITLE
Default cluster values configmap and kubeconfig

### PR DESCRIPTION
**Pull request by rossf7**
Towards https://github.com/giantswarm/giantswarm/issues/11656

This PR defaults the cluster values configmap and kubeconfig. Next PR will default the labels.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -4,12 +4,13 @@ go 1.15
 
 require (
 	github.com/giantswarm/apiextensions/v3 v3.7.0
-	github.com/giantswarm/app/v3 v3.1.1
+	github.com/giantswarm/app/v3 v3.1.2-0.20201111101054-8e371e128a01
```
Comment (rossf7) [inline go.mod:7]: TODO: Use released app library.

**Comment by rossf7** [comment]
Tested on `ginger`.

File: pkg/app/mutate_app.go [hunk 2]
```diff
@@ -61,18 +63,92 @@ func (m *Mutator) Mutate(request *v1beta1.AdmissionRequest) ([]mutator.PatchOper
...
+	}
+
+	result = append(result, mutator.PatchAdd("/spec/config/configMap", map[string]string{}))
+	result = append(result, mutator.PatchAdd("/spec/config/configMap/namespace", appNewCR.Namespace))
+	result = append(result, mutator.PatchAdd("/spec/config/configMap/name", key.ClusterConfigMapName(appNewCR)))
+
+	return result, nil
+}
+
+func (m *Mutator) mutateKubeConfig(ctx context.Context, appNewCR, appOldCR v1alpha1.App) ([]mutator.PatchOperation, error) {
```
Comment (tomahawk28) [inline pkg/app/mutate_app.go:131]: I can't find any codes using `appOldCR`, can we delete it?

File: pkg/app/mutate_app.go [hunk 3]
```diff
@@ -61,18 +63,92 @@ func (m *Mutator) Mutate(request *v1beta1.AdmissionRequest) ([]mutator.PatchOper
...
+	}
+
+	return result, nil
 }
 
 func (m *Mutator) Resource() string {
 	return Name
 }
+
+func (m *Mutator) mutateConfig(ctx context.Context, appNewCR, appOldCR v1alpha1.App) ([]mutator.PatchOperation, error) {
```
Comment (tomahawk28) [inline pkg/app/mutate_app.go:111]: Here too, I can't find `appOldCR` usage.

File: pkg/app/mutate_app.go [hunk 4]
```diff
@@ -61,18 +63,92 @@ func (m *Mutator) Mutate(request *v1beta1.AdmissionRequest) ([]mutator.PatchOper
...
+	if key.AppConfigMapName(appNewCR) != "" || key.AppConfigMapNamespace(appNewCR) != "" {
+		return nil, nil
+	}
+
+	// If there is no secret then create a patch for the config block.
+	if key.AppSecretName(appNewCR) == "" && key.AppSecretNamespace(appNewCR) == "" {
+		result = append(result, mutator.PatchAdd("/spec/config", map[string]string{}))
+	}
+
+	result = append(result, mutator.PatchAdd("/spec/config/configMap", map[string]string{}))
```
Comment (tomahawk28) [inline pkg/app/mutate_app.go:124]: What if we just add `namespace` and `name` from the struct in here?

**Review by tomahawk28 — COMMENTED** [review]
Looks good, just a few question.

File: pkg/app/mutate_app.go (hunk 3 above)
Comment (rossf7) [reply pkg/app/mutate_app.go:111]: Yes I'll remove it. The admission request has both but we don't need the old CR version.

File: pkg/app/mutate_app.go (hunk 4 above)
Comment (rossf7) [reply pkg/app/mutate_app.go:124]: Nice idea! I'll try that.

Comment (rossf7) [reply pkg/app/mutate_app.go]: Much better! Changes made. @tomahawk28 PTAL

**Review by tomahawk28 — APPROVED** [review]
LGTM!

**Pull request merged**